### ISSUE TITLE
Image Segementation Keras running too much slow

### ISSUE THREAD

**Issue by fizaict** [divamgupta/image-segmentation-keras#81]
I am new to nueral networks and I apologize if I say something wrong. I want to implement image segmentation using Keras. For this I am following this repository

https://github.com/divamgupta/image-segmentation-keras

I am trying to train my model using vgg_unet. This is the python script that I have executed

    import keras_segmentation

    model = keras_segmentation.models.unet.vgg_unet(n_classes=51 ,  input_height=416, 
    input_width=608  )

     model.train( 
       train_images =  "dataset1/images_prepped_train/",
       train_annotations = "dataset1/annotations_prepped_train/",
       checkpoints_path = "/tmp/vgg_unet_1" , epochs=5
     )

     out = model.predict_segmentation(
       inp="dataset1/images_prepped_test/0016E5_07963.png",
       out_fname="outp.png"
    )

    import matplotlib.pyplot as plt
    plt.imshow(out)

But it is working too much slow. I am using the Keras version 2.2.4. Kindly tell me that how can I execute it faster. Currently it is taking

    1/512 [..............................] - ETA: 2:52:50 - loss: 0.1299 - acc: 0. 

Can any one tell that how to speed up the execution? Please help me to resolve the issue.

**Comment by divamgupta** [comment]
Looks like you are training it on a CPU.  I would suggest you to use a GPU or a TPU to train.

**Comment by fizaict** [comment]
Will you please tell that from where I can get access to TPU? Your help will be really appreciated
!

**Comment by fizaict** [comment]
@divamgupta will you please tell how can I get access to TPU? Also how can I run it on TPU? Your help will be appreciated please help me

**Comment by divamgupta** [comment]
You can get it on google cloud. https://cloud.google.com/tpu/

**Comment by fizaict** [comment]
@divamgupta will you please tell how can i run it on tpu? Please i really need that help

**Comment by sainatarajan** [comment]
@fizaict Just run your code in Google colab. Before running, click on runtime and select change runtime type to TPU or GPU.
